var/spack/repos/builtin/packages/tl-expected/package.py

```diff
@@ -15,13 +15,15 @@ class TlExpected(CMakePackage):
 
     maintainers("charmoniumQ")
 
-    license("CC0-1.0")
-
-    # Note that the 1.0.0 has this issue:
-    # https://github.com/TartanLlama/expected/issues/114
-    # But no new patch version has been released,
-    # so I will use the latest commit at the time of writing:
-    version("2022-11-24", commit="b74fecd4448a1a5549402d17ddc51e39faa5020c")
+    license("CC0-1.0", checked_by="wdconinc")
+
+    version("1.1.0", sha256="1db357f46dd2b24447156aaf970c4c40a793ef12a8a9c2ad9e096d9801368df6")
+    with default_args(deprecated=True):
+        # Note that the 1.0.0 has this issue:
+        # https://github.com/TartanLlama/expected/issues/114
+        # But no new patch version has been released,
+        # so I will use the latest commit at the time of writing:
+        version("2022-11-24", commit="b74fecd4448a1a5549402d17ddc51e39faa5020c")
```

Comment:
I'm not very comfortable with the ordering here since `2022-11-24` is technically newer wrt version comparison than `1.1.0`.  Did you intend to include `1.0.0` under the context manager?

Reply:
That's the problem here: the spack version ordering puts `2022-11-24` newer than `1.1.0` but it isn't. Version `1.1.0` was released on Mar 15, 2023. The actual time ordering in releases is as listed in the package now.

If you want, I can create a non-deprecated version `1.0.0-26-gb74fecd` that corresponds to the same version that is now called `2022-11-24` (and still keep the dated version deprecated).

I didn't want to deprecate `1.0.0`, so it is intentionally outside the context manager.

Reply:
@tldahlgren Gentle ping. Does the above explanation satisfy this thread?